bevy_lint/tests/ui/unit_in_bundle/spawn_empty.rs

```diff
@@ -0,0 +1,75 @@
+#![feature(register_tool)]
+#![register_tool(bevy)]
+#![deny(bevy::unit_in_bundle)]
+#![allow(dead_code)]
+
+use bevy::{
+    ecs::relationship::{RelatedSpawner, RelatedSpawnerCommands},
+    prelude::*,
+};
+
+#[derive(Component)]
+struct ComponentA;
+
+#[derive(Component)]
+struct ComponentB;
+
+pub fn test_commands(
+    mut commands_owned: Commands,
+    commands_borrowed: &mut Commands,
+    mut commands_boxed: Box<Commands>,
+) {
+    commands_owned.spawn(());
+    //~^ ERROR: created a `Bundle` containing a unit `()`
+    //~| HELP: `spawn_empty()` is more efficient
+    commands_borrowed.spawn(());
+    //~^ ERROR: created a `Bundle` containing a unit `()`
+    //~| HELP: `spawn_empty()` is more efficient
+    commands_boxed.spawn(());
+    //~^ ERROR: created a `Bundle` containing a unit `()`
+    //~| HELP: `spawn_empty()` is more efficient
```

Comment:
This is why we use `expr_ty_adjusted()`. If we didn't, the lint would see this method call as `Box::spawn()`[^1] instead of the actual `Commands::spawn()`.

[^1]: Technically `Box<World>::spawn()`, but the linter would see it as a different type from `Commands` either way.